### PR TITLE
Add stamp option to docker_bundle rule

### PR DESCRIPTION
**Pull request by ixdy**
This allows using workspace status information (such as git describe) in image tags. To differentiate from make variables (and laziness), I'm using python formatting syntax, e.g. `{STABLE_BUILD_USER}` or `{MY_CUSTOM_VAR}`.

This is basically an implementation of what I was describing in https://github.com/bazelbuild/rules_docker/issues/3#issuecomment-297563224.

I think it'd work for what I need for kubernetes right now, though I'm not sure if this is the correct implementation, e.g. does it makes sense to do the stamping in `docker_bundle`? Should it also be supported in `docker_build` and/or `docker_push`? What things should be "stampable"? Do we want to be able to embed stamping info in image labels? (I might in the future, e.g. embed revision/other build data.)

(`docker_push` actually looks harder, since it's basically just a shell script template wrapper to the code in containerregistry.)

The proper way to do all of this would be entirely in skylark, though that depends on  https://github.com/bazelbuild/bazel/issues/1054 being resolved.

### CODE REVIEW

File: docker/join_layers.py [hunk 1]
```diff
@@ -87,13 +90,24 @@ def main():
...
   tag_to_layer = {}
   layer_to_tags = {}
+  stamp_info = {}
+
+  if args.stamp_info_file:
+    for infofile in args.stamp_info_file:
+      with open(infofile) as info:
+        for line in info:
+          line = line.strip("\n")
+          key, value = line.split(" ")
```
Comment (mattmoor) [inline docker/join_layers.py]: you probably want `line.split(" ", 1)` (or maybe 2, I can never remember the sense of this)

File: docker/join_layers.py [hunk 2]
```diff
@@ -87,13 +90,24 @@ def main():
...
+          stamp_info[key] = value
```
Comment (mattmoor) [inline docker/join_layers.py]: Can you print a warning when `key` exists in the map?

**Comment by ixdy** [comment]
Thanks for the quick review! I've addressed your comments in a follow-up commit and can squash if desired.

**Comment by mattmoor** [comment]
GitHub will let me squash and merge, but I'm happy to leave the history there.  thanks for the change.

**Pull request merged**